src/turnstile.cpp

```diff
@@ -0,0 +1,369 @@
+			json payload = {
+				{"jsonrpc", "1.0"},
+				{"id", "someid"},
+				{"method", "getnewaddress"},
+			};
+			return payload;
+		},
+		[=] (QMap<double, json>* newAddrs) {
+			// Get block numbers
+			auto curBlock = Settings::getInstance()->getBlockNumber();
+			auto blockNumbers = getBlockNumbers(curBlock, curBlock + numBlocks, splits.size());
+
+			// Assign the amounts to the addresses.
+			QList<TurnstileMigrationItem> migItems;
+			
+			for (int i=0; i < splits.size(); i++) {
+				auto tAddr = newAddrs->values()[i].get<json::string_t>();
+				auto item = TurnstileMigrationItem { zaddr, QString::fromStdString(tAddr), destAddr,
+													 blockNumbers[i], splits[i], 
+													 TurnstileMigrationItemStatus::NotStarted };
+				migItems.push_back(item);
+			}
+
+			// The first migration is shifted to the current block, so the user sees something 
+			// happening immediately
+			if (migItems.size() == 0) {
+				// Show error and abort
+				QMessageBox::warning(mainwindow, 
+					"Locked funds", "Could not initiate migration.\nYou either have unconfirmed funds or the balance is too low for an automatic migration.");
+				return;
+			}
+
+			migItems[0].blockNumber = curBlock;
+
+			std::sort(migItems.begin(), migItems.end(), [&] (auto a, auto b) {
+				return a.blockNumber < b.blockNumber;
+			});		
+
+			writeMigrationPlan(migItems);
+			rpc->refresh(true);	// Force refresh, to start the migration immediately
+		}
+	);
+}
+
+
+QList<int> Turnstile::getBlockNumbers(int start, int end, int count) {
+	QList<int> blocks;
+	// Generate 'count' numbers between [start, end]
+	for (int i=0; i < count; i++) {
+		auto blk = (std::rand() % (end - start)) + start;
+		blocks.push_back(blk);
+	}
+
+	return blocks;
+}
+
+	// Need at least 0.0005 ZEC for this
+double Turnstile::minMigrationAmount = 0.0005;
+
+QList<double> Turnstile::splitAmount(double amount, int parts) {
+	QList<double> amounts;
+
+	if (amount < minMigrationAmount)
+		return amounts;
+	
+	fillAmounts(amounts, amount, parts);
+	//qDebug() << amounts;
+
+	// Ensure they all add up!
+	double sumofparts = 0;
+	for (auto a : amounts) {
+		sumofparts += a;
+	}
+	
+	// Add the Tx fees
+	sumofparts += amounts.size() * Utils::getMinerFee();
+
+	//qDebug() << QString::number(sumofparts, 'f', 8) << QString::number(amount, 'f', 8);
+	//Q_ASSERT(QString::number(sumofparts, 'f', 8) == QString::number(amount, 'f', 8));
+	return amounts;
+}
+
+void Turnstile::fillAmounts(QList<double>& amounts, double amount, int count) {
+	if (count == 1 || amount < 0.01) {
+		// Also account for the fees needed to send all these transactions
+		auto actual = amount - (Utils::getMinerFee() * (amounts.size() + 1));
+
+		amounts.push_back(actual);
+		return;
+	}
+
+	// Get a random amount off the amount (between half and full) and call recursively.
+	// Multiply by hundered, because we'll operate on 0.01 ZEC minimum. We'll divide by 100 later
+	double curAmount = std::rand() % (int)std::floor(amount * 100);
```

Comment:
This doesn't appear to be doing what the comment says ("between half and full").

Reply:
Thanks for looking at this. I really appreciate it. 

Just note that the latest code lives on master branch (I've incorporated a bunch of feedback there, so please look at turnstile.cpp) here: https://github.com/adityapk00/zec-qt-wallet/blame/master/src/turnstile.cpp

Reply:
Fixed in b150f074f7ad2baad3179510b5f716cc1ad79bb6